zrcs/zrc-3.md

```diff
@@ -0,0 +1,431 @@
+| @param | `tokenOwner` | `ByStr20` | Address holding the tokens to be burned.       |
+| @param | `amount`     | `Uint128` | Number of tokens to be burned.                 |
+
+|           | Name             | Description                 | Event Parameters                                                                                                                                                                                                                                              |
+| --------- | ---------------- | --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `TellerBurnSuccess` | Burning is successful.     | `teller`: `ByStr20`, `tokenOwner`: `ByStr20`, and `amount`: `Uint128`.                                                                                               |
+| eventName | `Error`          | Burning is not successful. | - emit `CodeNotAuthorised` if the transition is called by an teller who is not authorized. |
+
+
+#### 7. Mint
+
+```ocaml
+(* @dev: Mint new tokens. Only contractOwner can mint.                        *)
+(* @param recipient:     Address of the recipient whose balance is increased. *)
+(* @param amount:        Number of tokens to be burned.                       *)
+transition Mint(recipient: ByStr20, amount: Uint128)
+```
+
+|        | Name        | Type      | Description                                          |
+| ------ | ----------- | --------- | ---------------------------------------------------- |
+| @param | `recipient` | `ByStr20` | Address of the recipient whose balance is increased. |
+| @param | `amount`    | `Uint128` | Number of tokens to be minted.                       |
+
+|           | Name          | Description                | Event Parameters                                                                                                                                                                                                                   |
+| --------- | ------------- | -------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `MintSuccess` | Minting is successful.     | `recipient`: `ByStr20`, and `amount`: `Uint128`.                             |
+| eventName | `Error`       | Minting is not successful. | - emit `CodeTokenExists` if the token already exists.<br>- emit `CodeNotAuthorised` if the transition is called by a user who is not the contract owner.<br>**NOTE:** Only the `contractOwner` is allowed to call this transition. |
+
+
+#### 8. TellerMint
```

Comment:
Why do we want `TellerMint`? Normal users should never be allowed to mint tokens anyway.